packages/editor/src/components/post-actions/actions.js

```diff
@@ -1020,6 +1062,7 @@ export function usePostActions( { postType, onActionPerformed, context } ) {
 	);
 
 	const duplicatePostAction = useDuplicatePostAction( postType );
+	const trashPostAction = useTrashPostAction( postType );
```

Comment:
Why don't we check dynamically add the action here instead of trying to check for the capabilities in `isEligible`?

Reply:
Hi @ellatrix, because a user may be able to trash a post and not another one. The permissions depend on the specific post. For example, a user maybe have permissions to delete only drafs, or permission to delete drafs authored by that user or permissions to delete private posts but not public ones, etc... We need to check for the specific post in isEligible.